### PR TITLE
[pixman] Fix x86 builds not using MSVC

### PR DESCRIPTION
**Pull request by Flole998**
Fixes the `#Todo` comment in pixman's portfile.cmale

### CODE REVIEW

**Comment by MonicaLiu0311** [comment]
There are two situations when modifying files under `vcpkg/ports/portName`:
1. If a version is released upstream, update the `"version"` field in `vcpkg.json` and count `"port-version"` from 0 (port-version is a field that records changes specific to vcpkg, 0 generally means deleting "port-version" line). Reference: [38749/files](https://github.com/microsoft/vcpkg/pull/38749/files).
2. If the upstream has not released a new version, then after modifying other files, please accumulate the `"port-version"` value to record this modification. Reference: [37966/files](https://github.com/microsoft/vcpkg/pull/37966/files).

Please read the documentation: [maintainer guide#versioning](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning).

File: ports/pixman/portfile.cmake [hunk 1]
```diff
@@ -4,8 +4,12 @@ if(VCPKG_TARGET_IS_UWP)
             -Dsse2=disabled
             -Dssse3=disabled)
 elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-    set(VCPKG_CXX_FLAGS "/arch:SSE2 ${VCPKG_CXX_FLAGS}") # TODO: /arch: flag requires compiler check. needs to be MSVC
-    set(VCPKG_C_FLAGS "/arch:SSE2 ${VCPKG_C_FLAGS}")
+    if(VCPKG_DETECTED_CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+        set(VCPKG_C_FLAGS "/arch:SSE2 ${VCPKG_C_FLAGS}")
+    endif()
+    if(VCPKG_DETECTED_CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
```
Comment (dg0yt) [inline ports/pixman/portfile.cmake:10]: `VCPKG_DETECTED_CMAKE_C_COMPILER_ID`, `VCPKG_DETECTED_CMAKE_CXX_COMPILER_ID` are undefined at this point.

File: ports/pixman/portfile.cmake [hunk 2]
```diff
@@ -4,8 +4,12 @@ if(VCPKG_TARGET_IS_UWP)
...
+        set(VCPKG_CXX_FLAGS "/arch:SSE2 ${VCPKG_CXX_FLAGS}")
+    endif()
     list(APPEND OPTIONS
             -Dmmx=enabled
             -Dsse2=enabled
```
Comment (dg0yt) [inline ports/pixman/portfile.cmake:15]: I wonder if this isn't enough.

Comment (Flole998) [reply ports/pixman/portfile.cmake:15]: Apparently the missing flag breaks the build, so I can remove the CFLAGS on my next attempt and we can see if the CI runs through.

**Comment by MonicaLiu0311** [comment]
At present, the version information of your commit is inconsistent with your local one:
```
##[error]Found the following errors:
##[error]versions/p-/pixman.json(2,): error : while parsing versions for pixman from version/p-/pixman.json
0.43.4 is declared with f7e1a97a6af1ad657489f901a6ca14db24652eff, but the local port has a different SHA a4e66badb15b7db89e768b1a1943583bff608b2c.
```
Please run `./vcpkg x-add-version testPort [--overwrite-version]` before each push:
1. For any file modification under the `ports/testPort/` folder, please commit after modification
2. Run `./vcpkg x-add-version portName` to generate a series of version according to your commit info (Add `--overwrite-version` when generating other than the first time)
3. Commit verison information again
4. Push together

`--overwrite-version`
After the first modification, please run `./vcpkg x-add-version testPort` to generate the corresponding json; for subsequent modifications of the same version, please use `./vcpkg x-add-version portName --overwrite-version` to update git-tree.

[maintainer guide#versioning](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning).

**Comment by MonicaLiu0311** [comment]
Note: I will be converting your PR to draft status. When you're ready, please revert to "ready for review".

**Comment by Flole998** [comment]
@microsoft-github-policy-service agree company="Flole"

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**